File: Data Structures & Algorithms/duplicate-integer/submission-7.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> freq;
        for (int i : nums){
            if (freq.count(i)){
                return true;
            }
            freq[i] = 1;
        }
        return false;
    }
};

File: Data Structures & Algorithms/is-anagram/submission-7.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        sf = {}
        for i in s:
            sf[i] = sf.get(i, 0) + 1
        tf = {}
        for i in t:
            tf[i] = tf.get(i, 0) + 1
        return sf == tf